runtime-environment-spec/polkadot_re_spec.tm

```diff
@@ -1937,7 +1937,131 @@
+    block. Additionally, a separate Changes Trie is created for each child
+    storage where changes occure.
+
+    \;
+
+    Keys that were modified outside of extrinsics, such as changes made by
+    the Runtime directly, have a <verbatim|0xffffffff> value as it's indice.
+
+    \;
+
+    Those entries are generated when the Runtime calls
+    <verbatim|ext_storage_changes_root> as described in section
+    <reference|sect-ext-storage-changes-root>.
+  </definition>
+
+  <\definition>
+    <label|defn-storage-key-to-blocks>The <strong|storage key to blocks
+    mappings> track any changes which occured in a certain range of blocks.
+    The key is inserted into the Trie, where it's value is a SCALE encoded
+    array containing block numbers where extrinsics caused any changes to the
+    key. The block numbers are represented as unsigned 32-bit integers.
+
+    \;
+
+    Those entries are generated when the Runtime calls
+    <verbatim|ext_storage_changes_root> as described in section
+    <reference|sect-ext-storage-changes-root>, but unlike the other mappings,
+    they are not generated on each block. The Runtime sets the key
+    <verbatim|:changes_trie> to storage, where it's value is made out of two
+    concatenated 32-bit integers:
+
+    <\itemize-dot>
+      <item><verbatim|digest_interval> - The interval (in blocks) at which
+      block mappings are created. Block mappings are not created when this is
+      less or equal to <verbatim|1>.
+
+      <item><verbatim|digest_levels> - Maximal number of \Plevels\Q in the
+      hierarchy. <verbatim|0> means that block mappings are not created at
+      all. <verbatim|1> means only the regular <verbatim|digest_interval>
+      block mappings are created. Any other level means that the block
+      mappings are created every <math|digest_interval<rsup|level>> block for
+      each level in 1 to <verbatim|digest_levels>.
+    </itemize-dot>
+
+    The Polkadot node must track the interval and levels, verify if
+    conditions apply and then generate those mappings when the Runtime calls
+    <verbatim|ext_storage_changes_root>. The range spans from the last block
+    (where this kind of mappings were created) + 1 to the current block.
+  </definition>
+
+  <\definition>
+    <label|defn-storage-key-to-child-tries>The <strong|storage key to child
+    changes tries mappings> track any storage keys from child storages and
+    their corresponding Child Changes Trie. As described in Definition
+    <reference|defn-storage-key-to-extrinsics>, changes inside child storages
+    are inserted into their own Trie. The changed key is inserted into the
+    main Trie, where it's value is the Merkle proof as desribed in section
+    <reference|sect-merkl-proof> of the corresponding Child Storage Trie,
+    which stores the mappings between the storage key and exstrinsics.
+  </definition>
+
+  The Trie itself is not part of the block, but a separately maintained
```

Comment:
Do you mean the main trie or the changes trie here? Please clarify.

Reply:
The Changes Trie. Spec adjusted.